client/app/login/login.component.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { LoginComponent } from './login.component';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { MdInputModule, MdListModule } from '@angular/material';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import 'hammerjs';
+import { LoginService, MockLoginService } from '../shared/login.service';
+import { Router } from '@angular/router';
+
+describe('LoginComponent', () => {
+  let component: LoginComponent;
+  let fixture: ComponentFixture<LoginComponent>;
+
+  class MockRouter {
+    navigateByUrl = jasmine.createSpy('navigateByUrl');
+  }
+
+  let mockRouter: MockRouter;
+
+  beforeEach(async(() => {
+      mockRouter = new MockRouter(),
+      TestBed.configureTestingModule({
+      declarations: [ LoginComponent ],
+      imports: [
+        FormsModule,
+        ReactiveFormsModule,
+        MdInputModule,
+        BrowserAnimationsModule,
+        MdListModule
+      ],
+      providers: [{ provide: LoginService, useClass: MockLoginService }, { provide: Router, useValue: mockRouter }]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(LoginComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(LoginComponent);
+    component = fixture.componentInstance;
+    spyOn(component, 'login').and.callFake(() => {
+      return {token: 'token'};
+    });
+    fixture.detectChanges();
+  });
+
+  it('should be created', () => {
+    expect(component).toBeTruthy();
+  });
```

Comment:
Needs more test coverage. See [new-volunteer-form.component.spec.ts](https://github.com/cerberus-org/cerberus-app/blob/bf094cbb55917b43d0a490f95f49998ffb423c0d/client/app/volunteer-check-in/new-volunteer-form/new-volunteer-form.component.spec.ts) for examples.